x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/aggs/PCG.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.ml.aggs;
+
+import com.carrotsearch.hppc.BitMixer;
+
+/**
+ * Permuted congruential generator style random number generator
+ */
+public final class PCG {
+
+    private static final long MULTIPLIER = 6364136223846793005L;
+
+    private long state = 0x4d595df4d0f33173L;
+    private long increment = 1442695040888963407L;
+
+    public PCG(long stream) {
+        // The increment must be odd.
+        increment = stream << 1 | 1;
+    }
```

Comment:
Could this delegate to `PCG(0, stream)`? I'd feel a bit better as it's unclear to me why there are differences between these two ctors, e.g. why does the other ctor need to call nextInt() but not this one.